osu.Server.DifficultyCalculator/Program.cs

```diff
@@ -0,0 +1,214 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using McMaster.Extensions.CommandLineUtils;
+using MySql.Data.MySqlClient;
+using osu.Game.Beatmaps.Legacy;
+using osu.Game.Rulesets.Mania.Mods;
+using osu.Game.Rulesets.Mods;
+
+namespace osu.Server.DifficultyCalculator
+{
+    [Command]
+    public class Program
+    {
+        public static void Main(string[] args)
+            => CommandLineApplication.Execute<Program>(args);
+
+        // Todo: Move to configuration
+        private const string connection_string = "Server=localhost;Database=osu;User ID=root;SslMode=None;Pooling=true; Min pool size = 50; Max pool size = 200; Charset=utf8;";
```

Comment:
Definitely should be read from configuration file (check out `appsettings.json` documentation and implementation in https://github.com/ppy/osu-server/pull/1/files#diff-79f3b5ed3196a6a5ec6c0257b17fbbfa).

We may want to move this implementation to a shared project.

Reply:
Yeah not sure if a shared project is super necessary here, it seems like the only code that's really duplicated is:

```
var env = Environment.GetEnvironmentVariable("APP_ENV") ?? "development";
var config = new ConfigurationBuilder()
    .SetBasePath(Directory.GetCurrentDirectory())
    .AddJsonFile("appsettings.json", optional: true, reloadOnChange: false)
    .AddJsonFile($"appsettings.{env}.json", optional: true, reloadOnChange: false)
    .AddEnvironmentVariables()
    .Build();
```

Soo...

Reply:
Yeah, maybe not. I think @notbakaneko is planning on looking into whether there's an abstraction on top of appsettings to provide typing of settings to avoid string lookups.

Reply:
TOO LATE. We'll revise on this at a later point /shrug!